pageserver/src/tenant.rs

```diff
@@ -1269,6 +1281,18 @@ impl Tenant {
             }
         };
 
+        if let Some(remote_client) = loaded_timeline.remote_client.as_ref() {
+            // failing the await for uploads to complete will mean that the tenant is
+            // non-relocatable, however with erroring out it is clear to control plane that the
+            // request needs to be retried, and next time after restart they will get 409.
+            let kind = ancestor_timeline_id
+                .map(|_| "branched")
+                .unwrap_or("bootstrapped");
+            remote_client.wait_completion().await.with_context(|| {
```

Comment:
> Surfacing the possibility of that increasing startup time if console blocks creation on this request

It does block, cause we need tenant + timeline to start compute. So the sequence is trivial: create tenant -> create timeline (both in the same operation now) -> start compute. We probably also need to increase timeouts on these calls, previously we were expecting the pageserver API to be fast, but recently even getting timeline info takes >1s, so if we start blocking on S3 upload during creation, it could be much longer

Reply:
As you described above timeline id is known in advance, so you can start spinning up the compute pod and creating a tenant at the same time. Not sure if this is needed with pods pre creation in mind, but in current conditions it looks like a viable approach.

> but recently even getting timeline info takes >1s

Thats shouldnt be the case, do you have an example?

> so if we start blocking on S3 upload during creation, it could be much longer

Here I think we need to verify what is the impact. We can try to test this before merging, or merge it and check on staging. How much do we care about tenant initialization time? How much extra time we can afford here? It looks like this is not as important as cold start times, because this happens only once during creation. Additionally we can also have pool of pre created tenants that are not yet assigned to user's projects